src/Scene/AnimationPlayer.js

```diff
@@ -0,0 +1,231 @@
+const FRAMERATE = 60;
+const INTERVAL_TIME = 1000 / FRAMERATE;
+// if is true console.log are enabled to sniff animation'state
+const debugAnimation = false;
+
+// player statut
+const PLAYER_STATE = {
+    // player is stopped
+    STOP: 0,
+    // player plays animation
+    PLAY: 1,
+    // player is at the end of an animation
+    END: 2,
+    // player is paused
+    PAUSE: 3,
+};
+
+const debugMsg =
+    [
+        'Stop',
+        'Play',
+        'End',
+        'Pause',
+    ];
+
+// if debugAnimation is true console.log are enabled to sniff animation'state
+let _DEBUG = null;
+
+if (debugAnimation) {
+    _DEBUG = function DEBUG(message, animation) {
+        if (animation) {
+            // eslint-disable-next-line no-console
+            console.info('Animation ', message, ' : ', animation.name);
+        }
+    };
+} else {
+    _DEBUG = function _DEBUG() {};
+}
+
+// Private functions
+// stop timer and re-init parameter
+const resetTimer = function resetTimer(player) {
+    if (player.id) {
+        clearInterval(player.id);
+        player.id = undefined;
+    }
+    if (player.waitTimer) {
+        clearInterval(player.waitTimer);
+        player.waitTimer = undefined;
+    }
+    player.keyframe = 0;
+};
+
+// finish animation and re-init parameter
+const finishAnimation = function finishAnimation(player) {
+    resetTimer(player);
+    player.animation = null;
+    player.callBackFrame = () => {};
+    if (player.resolve) {
+        player.resolve();
+        player.resolve = null;
+        player.promise = null;
+    }
+};
+
+const setPlayerState = function setPlayerState(player, state) {
+    player.state = state;
+    _DEBUG(debugMsg[state], player.animation);
+};
+
+const frameEvent = new CustomEvent('frameAnimation');
+const stopEvent = new CustomEvent('stopAnimation');
+const endEvent = new CustomEvent('endAnimation');
+
+/**
+ * AnimationPlayer
+ * It can play, pause or stop Animation or AnimationExpression (See below).
+ * AnimationPlayer is needed to use Animation or AnimationExpression
+ * AnimationPlayer emits events :
+ *       - for each animation's frame;
+ *       - when Animation is stopped
+ *       - when Animation is ending
+ */
+class AnimationPlayer {
+    constructor(dom) {
+        this.dom = dom;
+        this.id = null;
+        this.keyframe = 0;
+        this.animation = null;
+        this.resolve = null;
+        this.promise = null;
+        this.state = PLAYER_STATE.STOP;
+        this.callBackFrame = () => {};
+        this.waitTimer = null;
+    }
+
+    /**
+     * Gets the interval time.
+     *
+     * @return {Number}  The interval time is the time between consecutive frames
+     */
+    static get intervalTime() {
+        return INTERVAL_TIME;
```

Comment:
* `frameDuration` would be a better name
* you don't know the framerate so assuming it'll be 60 FPS might be wrong...
* last but not least, this function is unused so you could remove it